src/main/resources/assets/js/components/AppComponent.jsx

```diff
@@ -3,8 +3,15 @@
 var React = require("react/addons");
 
 var AppComponent = React.createClass({
+  propTypes: {
+    router: React.PropTypes.object.isRequired
+  },
+
   onClick: function () {
-    this.props.onClick(this.props.model);
+    this.props.router.navigate(
+      "#apps" + this.props.model.get("id"),
```

Comment:
Shouldn't the routes be following as closely as possible to a REST style interface? I'd expect to see something more like:

`apps/:id`